addons/account_edi/models/account_edi_format.py

```diff
@@ -506,21 +506,90 @@ def _retrieve_partner(self, name=None, phone=None, mail=None, vat=None):
         :param vat:     The vat number of the partner.
         :returns:       A partner or an empty recordset if not found.
         '''
-        domains = []
-        for value, domain in (
-            (name, [('name', 'ilike', name)]),
-            (phone, expression.OR([[('phone', '=', phone)], [('mobile', '=', phone)]])),
-            (mail, [('email', '=', mail)]),
-            (vat, [('vat', 'like', vat)]),
-        ):
-            if value is not None:
-                domains.append(domain)
-
-        domain = expression.AND([
-            expression.OR(domains),
-            [('company_id', 'in', [False, self.env.company.id])],
-        ])
-        return self.env['res.partner'].search(domain, limit=1)
+        def search_with_vat(extra_domain):
+            if not vat:
+                return None
+
+            # Sometimes, the vat is specified with some whitespaces.
+            normalized_vat = vat.replace(' ', '')
+
+            partner = self.env['res.partner'].search([('vat', '=', normalized_vat)], limit=1)
+
+            # Try to remove the country code prefix from the vat.
+            if not partner and len(normalized_vat) > 2 and normalized_vat[:2].isalpha():
+                country_prefix = normalized_vat[:2]
+                normalized_vat = normalized_vat[2:]
+                partner = self.env['res.partner'].search(extra_domain + [
+                    ('vat', '=', normalized_vat),
+                    ('country_id.code', '=', country_prefix.lower()),
+                ], limit=1)
+
+                # The country could be not specified.
+                if not partner:
```

Comment:
Having the country code not being a valid country and the country code not being specified are not the same things. I'm not sure which one you want but this comment is confusing

Reply:
"The country code could be not specified on the VAT number" ?